app/src/main/java/com/babylon/wallet/android/presentation/dapp/unauthorized/login/DAppUnauthorizedLoginViewModel.kt

```diff
@@ -73,37 +92,86 @@ class DAppUnauthorizedLoginViewModel @Inject constructor(
-                    _state.update { it.copy(dapp = dApp) }
+
+            // check if request contains proofOfOwnershipRequestItem
+            // and if yes then validate the requested entities from this request item
+            val proofOfOwnershipRequestItem = request.proofOfOwnershipRequestItem
+            if (proofOfOwnershipRequestItem != null) {
+                val areRequestedAddressesValid = validateRequestedAddressesForProofOfOwnership(
+                    requestedPersonaAddress = proofOfOwnershipRequestItem.personaAddress,
+                    requestedAccountAddresses = proofOfOwnershipRequestItem.accountAddresses
+                )
+                if (areRequestedAddressesValid) {
+                    _state.update { state ->
+                        state.copy(
+                            personaRequiredProofOfOwnership = proofOfOwnershipRequestItem.personaAddress,
+                            accountsRequiredProofOfOwnership = proofOfOwnershipRequestItem.accountAddresses
+                        )
+                    }
+                } else {
+                    handleRequestError(RadixWalletException.DappRequestException.InvalidPersonaOrAccounts)
+                    return@launch
                 }
-            }.onFailure { error ->
-                _state.update { it.copy(uiMessage = UiMessage.ErrorMessage(error)) }
             }
+
             setInitialDappLoginRoute()
         }
     }
 
-    fun dismissNoMnemonicError() {
-        _state.update { it.copy(isNoMnemonicErrorVisible = false) }
+    private suspend fun getDappDetails(dAppDefinitionAddress: AccountAddress) {
+        stateRepository.getDAppsDetails(
+            definitionAddresses = listOf(dAppDefinitionAddress),
+            isRefreshing = false
+        ).onSuccess { dApps ->
+            dApps.firstOrNull()?.let { dApp ->
+                _state.update { it.copy(dapp = dApp) }
+            }
+        }.onFailure { error ->
+            _state.update { it.copy(uiMessage = UiMessage.ErrorMessage(error)) }
+        }
+    }
+
+    private suspend fun validateRequestedAddressesForProofOfOwnership(
+        requestedPersonaAddress: IdentityAddress?,
+        requestedAccountAddresses: List<AccountAddress>?
+    ): Boolean {
+        // if requestedPersonaAddress is present, then check if it exists in the allPersonasOnCurrentNetwork
+        val isIdentityValid = requestedPersonaAddress?.let { address ->
+            getProfileUseCase().allPersonasOnCurrentNetwork.any { it.address == address }
+        } ?: true // if requestedPersonaAddress is null, assume it's valid (no need to check)
+
+        // if requestedAccountAddresses are present, then check if all of them exist in the existingAccounts
+        val areAccountsValid = requestedAccountAddresses?.let { addresses ->
+            addresses.all { address ->
+                getProfileUseCase().allAccountsOnCurrentNetwork.any { it.address == address }
```

Comment:
The same question applies for not hidden accounts too.

Reply:
If all you need is the non hidden accounts you could use the method `activeAccountOnCurrentNetwork`.